contents/tutorials/react-surveys.md

```diff
@@ -0,0 +1,504 @@
+
+```bash
+npm install posthog-js
+```
+
+Once installed, import PostHog into `src/index.js` and set up a client using a project API key and instance address from the [project settings](https://app.posthog.com/project/settings). Then, wrap the app with `PostHogProvider` to access PostHog in any component.
+
+```js
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import './index.css';
+import App from './App';
+import posthog from 'posthog-js'
+import { PostHogProvider } from 'posthog-js/react'
+
+posthog.init('<ph_project_api_key>', {
+  api_host: '<ph_instance_address>',
+})
+
+const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+  <React.StrictMode>
+    <PostHogProvider client={posthog}>
+      <App />
+    </PostHogProvider>
+  </React.StrictMode>
+);
+```
+
+With PostHog set up, our React app is ready for the survey.
+
+## Creating a survey
+
+There are two options for displaying a survey using PostHog:
+
+1. Implement your own survey UI.
+2. Use PostHog's prebuilt survey UI.
+
+This tutorial will cover how to implement both options:
+
+## Option 1: Implement your own survey UI
+
+First, create a survey in PostHog by going to the [surveys tab](https://app.posthog.com/surveys) and clicking "New survey." Then, set up your survey with the following:
+
+1. Add a name (like `my-first-survey`)
+2. Set the display mode to `API`
+3. Set the question type to `Rating`, question title to `How likely are you to recommend us to a friend?`, display type to `number` and scale to `1-10`.
+4. You can leave the remaining optional properties blank (such as `Targeting` or `Thank you message`).
+
+Click "Save as draft" and then on the next screen click "Launch". We're now ready to integrate this survey into our app.
+
+![Custom survey set up](../images/tutorials/react-surveys/create-api-survey.png)
+
+### Adding the surveys code
+
+There are three parts to adding code for our custom survey:
+
+1. Create the survey UI.
+2. Add the logic for displaying it.
+3. Capture interactions from it.
+
+#### 1. Create the survey UI
+
+We've created an example survey UI for this tutorial. To use it, create a new file in `./src` folder called `Survey.js` and paste the following code:
+
+```js
+// src/Survey.js
+import React from 'react';
+
+function Survey({ title, onDismiss, onSubmit }) {
+  const [selectedValue, setSelectedValue] = React.useState(null);
+
+  const handleSelect = (value) => {
+    setSelectedValue(value);
+  }
+
+  const handleSubmit = () => {
+    onSubmit(selectedValue);
+  }
+
+  return (
+    <div className="survey-popup">
+      <h2>{title}</h2>
+      <div>
+        {[...Array(10)].map((_, i) => (
+          <button key={i + 1} onClick={() => handleSelect(i + 1)}>{i + 1}</button>
+        ))}
+      </div>
+      <div>
+        <button onClick={onDismiss}>Dismiss</button>
+        <button onClick={handleSubmit}>Submit</button>
```

Comment:
Could we remove this button and just make selecting value dismiss the survey? Kind of what I expected it to do.